src/function/gds/all_shortest_paths.cpp

```diff
@@ -124,19 +124,23 @@ class AllSPDestinationsOutputWriter : public DestinationsOutputWriter {
         processor::NodeOffsetMaskMap* outputNodeMask)
         : DestinationsOutputWriter{context, rjOutputs, outputNodeMask} {}
 
-    std::unique_ptr<RJOutputWriter> copy() override {
-        return std::make_unique<AllSPDestinationsOutputWriter>(context, rjOutputs, outputNodeMask);
-    }
-
-protected:
-    void writeMoreAndAppend(processor::FactorizedTable& fTable, common::nodeID_t dstNodeID,
-        uint16_t) const override {
-        auto multiplicity =
-            rjOutputs->ptrCast<AllSPDestinationsOutputs>()->multiplicities.getTargetMultiplicity(
-                dstNodeID.offset);
-        for (uint64_t i = 0; i < multiplicity; ++i) {
+    void write(FactorizedTable& fTable, nodeID_t dstNodeID,
+        processor::RJOutputCounter* counter) override {
+        auto outputs = rjOutputs->ptrCast<AllSPDestinationsOutputs>();
+        auto length = outputs->pathLengths->getMaskValueFromCurFrontierFixedMask(dstNodeID.offset);
+        dstNodeIDVector->setValue<nodeID_t>(0, dstNodeID);
```

Comment:
Was this a bug? Why were these code not there before?

Reply:
Nah I simply removed one virtual function and choose to duplicate several lines of code.